.github/workflows/backend-typing.yml

```diff
@@ -0,0 +1,82 @@
+name: backend typing
+on:
+  push:
+    branches:
+      - master
+      - releases/**
+  pull_request:
+
+jobs:
+  test:
+    name: backend typing
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        instance: [0]
+
+    steps:
+      - uses: actions/checkout@v2
+
+        with:
+          # Avoid codecov error message related to SHA resolution:
+          # https://github.com/codecov/codecov-bash/blob/7100762afbc822b91806a6574658129fe0d23a7d/codecov#L891
+          fetch-depth: '2'
+
+      # If we make these jobs "required" to merge on GH, then on every PR, GitHub automatically
+      # creates a status check in the "pending" state. This means that the workflow needs to run
+      # for every PR in order to update the status checks.
+      #
+      # In order to optimize CI usage, we want the tests to only run when python files change,
+      # since frontend changes should have no effect on these test suites. We cannot use GH workflow
+      # path filters because entire workflow would be skipped vs skipping individual jobs which
+      # would still allow this status check to pass.
+      - name: Check for python file changes
+        uses: getsentry/paths-filter@v2
+        id: changes
+        with:
+          token: ${{ github.token }}
+          filters: .github/file-filters.yml
```

Comment:
It'd be nice if we could skip the rest of the steps if no files specified in mypy were modified

Reply:
That's a good idea. If you have suggestions of other unnecessary steps to trim from this, that would be helpful too. It seems like a lot of steps to run mypy.

Reply:
Do you have a suggestion of how to avoid duplicating the filenames in the file-filters as well as mypy.ini? I'm trying to make typing as frictionless as possible and it would be nice if people only had to update one file.

Reply:
Hmm I don't have any suggestions yet, let's not block on this, instead maybe add a ticket for @getsentry/productivity to look into it in the future. If this workflow runs fast enough it shouldn't be a big deal.

Reply:
How do I add a ticket?

Reply:
https://github.com/getsentry/sentry/issues/new/choose :D - in `Projects` on the RHS, add it to CI/tooling

Reply:
Did you get to this?

Reply:
I didn't want to add it until I actually had something in Sentry. Seemed weird to make a feature request for a feature that didn't exist yet. I planned to open a ticket once this was approved/merged.